### ISSUE TITLE
"[expression.unparsable.type.invalid] Expression invalid in dependent type annotation" in unannotated code

### ISSUE THREAD

**Issue by kennknowles** [typetools/checker-framework#3792]
Steps to reproduce:

```
git clone --depth 1 --branch checker-curiosity https://github.com/kennknowles/beam
cd beam
./gradlew compileJava
```

Output:

```
/Users/klk/GitHub/apache/beam/runners/core-java/src/main/java/org/apache/beam/runners/core/InMemoryStateInternals.java:504: error: [expression.unparsable.type.invalid] Expression invalid in dependent type annotation: [error for expression: this.contents.subMap(minTimestamp, true, limitTimestamp, false); error: Invalid 'true' because is not a supported expression]
          .flatMap(e -> e.getValue().stream().map(v -> TimestampedValue.of(v, e.getKey())))
                   ^
```

Since this is unannotated code, I do not really know how to interpret this error message.

**Comment by smillst** [comment]
This warning is the result of a bug in the Checker Framework.  I've fixed it in #3805 and confirmed that the fix works on beam.

**Comment by kennknowles** [comment]
Thank you!
